src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantInfo.hpp

```diff
@@ -42,8 +42,9 @@ class DiscoveryParticipantInfo : public DiscoverySharedInfo
 public:
 
     DiscoveryParticipantInfo(
-            eprosima::fastrtps::rtps::CacheChange_t* change_)
-        : DiscoverySharedInfo(change_)
+            eprosima::fastrtps::rtps::CacheChange_t* change,
+            const eprosima::fastrtps::rtps::GuidPrefix_t& guid_p)
```

Comment:
I change the name, but including a vector here would lead in an efficiency lack: every time an entity is added will create a new vector with just one value and will destroy it afterwards.
Probably both constructors should exist in a future step, but I would rather say to implement it when it is needed.